### PR TITLE
nbpackage attempt

### PR DESCRIPTION
**Pull request by ebarboni**
-- try not to put on nighlies and archivedartefact the old linux installer

then unpack bin;zip from rao snapshot to be able to use experimental until proper release (22.04 compression fix need to be done in nbpackage)

try to build installer.

@neilcsmith-net put you reviewer so you can merge once rc2 in ok at the pace you want.

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
Thanks.  Will take a look shortly.  Do you have any sort of  test setup for this?

Have handled locally manually and pushed to https://nightlies.apache.org/netbeans/candidate/installers/ for rc2 anyway.  Also deleted old installer.

**Comment by ebarboni** [comment]
Only test partial on a linux box but not on a spare jenkins . But once rc2 is locked and this go in I can patch and check

only deb as https://ci-builds.apache.org/job/Netbeans/job/testnode/4/console which only show the deb up for now

**Comment by neilcsmith-net** [comment]
OK.  Well, it's locked and gone!  Does dependency unpack handle the exec permission OK?

In terms of the deb compression, I currently have a `dpkg-deb` script ahead of the original in my path which contains -

```
#!/usr/bin/env sh

echo "WARNING : OVERRIDDEN DPKG_DEB"

exec /usr/bin/dpkg-deb -Zxz "$@"

```

If we could replicate that, we wouldn't need the immediate fix in NBPackage.

File: vars/asfMainNetBeansBuild.groovy [hunk 1]
```diff
@@ -371,13 +355,26 @@ def doParallelClusters(cconfigs) {
...
+                                    {
+                                        // unpack nbpackage snapshot can later
+                                        sh "mvn org.apache.maven.plugins:maven-dependency-plugin:3.5.0:unpack -DoutputDirectory=nbpackage${versionnedpath}installer -Dartifact=org.apache.netbeans:nbpackage:${nbpackageversion}:bin:zip -Dmaven.repo.local=${env.WORKSPACE}/.repository -DremoteRepositories=apache.snapshots.https::::https://repository.apache.org/snapshots"
+                                        
+                                        // build installer only deb for testing.
+                                        sh "nbpackage${versionnedpath}installer/nbpackage-${nbpackageversion}/bin/nbpackage --type linux-deb -Pname=\"Apache NetBeans\" -Pversion=${version} -Purl=\"https://netbeans.apache.org\" -Pdeb.maintainer=\"NetBeans Mailing List <users@netbeans.apache.org>\" -Pdeb.desktop-filename=\"apache-netbeans-ide-${version}\" -Pdeb.wmclass=\"Apache NetBeans IDE ${version}\" --input dist${versionnedpath}${path}-${rmversion}-bin.zip"
```
Comment (neilcsmith-net) [inline vars/asfMainNetBeansBuild.groovy:374]: `deb.desktop-filename` and `deb.wmclass` should probably be using `${rmversion}` here.  They need to include any suffix in the main window title.  `-Pversion` would be good as `${rmversion}` too probably, although using a hyphen breaks the RPM packager currently (hence 17~rc2 on nightlies).

**Comment by ebarboni** [comment]
unpack give me the execute locally so maybe. (but big cliffhanger)

Just commented the publishtonightlies to no interfer

File: vars/asfMainNetBeansBuild.groovy (hunk 1 above)
Comment (ebarboni) [reply vars/asfMainNetBeansBuild.groovy]: put in multiline string as it's unreadble.

groovy seems ok in my local jenkins but cannot go far

**Comment by ebarboni** [comment]
@neilcsmith-net I fix an issue with locatRepo not defined by testing locally. I had to add withAnt to be able to reproduce upon mvn dep get. 

I need to be live to see if it can go after the mvn.

**Comment by neilcsmith-net** [comment]
Well, we've got time to test and revert if necessary before rc3.  Although with rpm and deb compression options currently missing, we might not be able to use this for the release anyway, so could test after 17 is out?  Your call!

It would be good to remove the whitespace changes.  I find it hard to read what's actually changed in the diff.

**Comment by ebarboni** [comment]
well the withAnt addition is the most annoying. Btw need to chek if nbpackage is reached. 
will try the deb fix also later.

let's try

**Pull request merged**